RTLA documentation (#100)
canonical/real-time-ubuntu-docs: docs/reference/real-time-metrics-tools.rst

This command displays system resource statistics with timestamps and shows most frequent interrupts.

rtla
-----

> `RTLA`_ provides a set of tools for the analysis of the kernel's realtime behavior on specific hardware.

Reviewer: ```suggestion
`RTLA`_ provides a set of tools for the analysis of the kernel's real-time behavior on specific hardware.
```

RTLA comes pre-installed on Ubuntu 24.04 LTS (Noble Numbat) and later. 
It is not available as a package on Ubuntu Core.

hwnoise
^^^^^^^

> See how hardware noise affects kernel threads.

Reviewer: ```suggestion

See how hardware noise affects kernel threads.
```

hwnoise is similar to osnoise, except that it disables interrupts. Only non-maskable interrupts and hardware-related noise will affect the test.

Run ``rtla hwnoise top`` for a top-like interface that shows the time each thread takes to read the current time.

Run ``rtla hwnoise hist`` for a histogram, which may be better suited for logging.

For more information, see the `rtla-hwnoise`_ documentation.

osnoise
^^^^^^^

> See how operating system noise (preemption, IRQs/soft-IRQs) affect kernel threads.

Reviewer: ```suggestion

See how operating system noise (preemption, IRQs/soft-IRQs) affect kernel threads.
```

> osnoise is part of the `rtla`_ suite, which comes pre-installed on Ubuntu 24.04 LTS (Noble Numbat) and later. 

Reviewer: ```suggestion
`osnoise` is part of the `rtla`_ suite, which comes pre-installed on Ubuntu 24.04 LTS (Noble Numbat) and later. 
```

It does not support older versions of Ubuntu.

Run ``rtla osnoise top`` for a top-like interface that shows the time each thread takes to read the current time.

Run ``rtla osnoise hist`` for a histogram, which may be better suited for logging.

For more information, see the `rtla-osnoise-top`_ and `rtla-osnoise-hist`_ documentation.

> 

Reviewer: ```suggestion
```

timerlat

> ^^^^^^^^

Reviewer: ```suggestion
^^^^^^^^

```

> timerlat runs threads which periodically wake up and sleep, and data is collected about how much latency overhead the operating system adds to them.

Reviewer: ```suggestion
`timerlat` runs threads which periodically wake up and sleep, and data is collected about how much latency overhead the operating system adds to them.
```

> Like osnoise, timerlat offers both a ``timerlat top`` and ``timerlat hist`` mode.

Reviewer: ```suggestion
Like ``osnoise``, ``timerlat`` offers both a ``timerlat top`` and ``timerlat hist`` mode.
```

For more information, see the `rtla-timerlat-top`_ and `rtla-timerlat-hist`_ documentation.

> 

Reviewer: ```suggestion
```

This command runs the ``oslat`` test and then prints the results, which include latency measurements for each core.

RTLA
----
`RTLA`_ provides a set of tools for the analysis of the kernel’s realtime behavior on specific hardware.

RTLA comes pre-installed on Ubuntu 24.04 LTS (Noble Numbat) and later. 

> It does not support older versions of Ubuntu.

Reviewer 2: What about Ubuntu Core?

Reviewer 2: I believe it should come from the base. If you have core24 snap installed, e.g. looking for `ps`:
```
find /snap/core24/current/bin/ -name "*ps"
```

osnoise, etc aren't there.

Another way to verify is to spin up a multipass UC24 VM and look what is available inside:
```shell
multipass launch core24 --name=test-uc24
multipass exec test-uc24 bash
```

Since the documentation is for both classic and core, it should explain if the command isn't available on one. Please see other commands documented there and try to maintain the consistency (description, example, etc).

Author: Okay, I tried both methods, and I don't see it in core24 snap. Does this mean it isn't available for Ubuntu core? Or that you'd just have to manually install it?

Reviewer 2: Correct, it is not available. We just need to mention it in the doc that it is not available on Ubuntu Core.

This command runs the ``oslat`` test and then prints the results, which include latency measurements for each core.

> 

Reviewer: ```suggestion
```
